### PR TITLE
feat(ssr): add server side rendering with graphql API

### PR DESCRIPTION
**Pull request by amille44420**

### CODE REVIEW

File: api/index.jsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+server.use(history());
+
+// serve it
+server.get('*', async (req, res, next) => {
+    try {
+        // create the apollo client
+        const apolloClient = createApolloClient(generateApolloSettings(req));
+        // we don't have the redux store yet
+        // create a callback to retrieve it
+        let store = (instance) => {
```
Comment (jvasseur) [inline api/index.jsx]: I'm not found of using the same variable to store the store and the callback to get it back. This makes the variable change it's significance base on the state of the app.

File: api/loaders/index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+const initializers = Object.entries({
+    usersById,
+});
+
+const initLoaders = (...args) => initializers.reduce(
+    (loaders, [key, init]) => ({
+        ...loaders,
+        [key]: init(...args),
+    }),
```
Comment (jvasseur) [inline api/loaders/index.js]: I'm not found of this piece of code, it's not clear at first glance what it does.

We could either:
 - Add ramda in the project to use it's map method
 - Use `fromEntries` (we can use it now that we require node 12) in a `entries` -> `map` -> `fromEntries` chain.

File: api/loaders/utils/loadById.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import { Op } from 'sequelize';
+
+const loadById = (model) => () => {
+    const batchLoader = async (ids) => {
+        const instances = await model.findAll({ where: { id: { [Op.in]: ids } } });
+
+        const idMap = instances.reduce((acc, instance) => ({
+            ...acc,
+            [instance.id]: instance,
+        }), {});
```
Comment (jvasseur) [inline api/loaders/utils/loadById.js]: Same here, we could use `fromEntries`.

File: api/index.jsx (hunk 1 above)
Comment (amille44420) [reply api/index.jsx]: it's only for the SSR rendering, we need to retrieve the actual store to get its state and give it back to the client.

File: api/schema/mutations/createUser.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import bcrypt from 'bcrypt';
+import { User } from '@api/models';
+import { BookType } from '../types';
+
+export default {
+    args: {
+        email: { type: new NonNull(StringType) },
+        password: { type: new NonNull(StringType) },
+    },
+    type: new NonNull(BookType),
```
Comment (RomualdGeairon) [inline api/schema/mutations/createUser.js]: UserType

**Pull request merged**